Drivers/ECommerceCurrencySiteSettingsPartDriver.cs

```diff
@@ -23,7 +26,7 @@ protected override string Prefix
         }
         
         protected override DriverResult Editor(ECommerceCurrencySiteSettingsPart part, dynamic shapeHelper) {
-            if (_selectedCurrencyProvider is UseCurrencyFromSiteSettingsProvider) {
+            if (_currencyProvider is UseCurrencyFromSiteSettingsProvider) {
```

Comment:
I see your point. You are correct. My concern here was the case when someone would be enabling a different feature giving a different provider, But even if they did, that provider would not (should not) be using the setting here.